userspace/falco/app_actions/configure_interesting_sets.cpp

```diff
@@ -15,27 +15,106 @@ limitations under the License.
 */
 
 #include "application.h"
+#include "configure_interesting_sets.h"
+#include <unordered_set>
+#include <sinsp.h>
+#include <sstream>
 
 using namespace falco::app;
 
-void application::configure_interesting_sets()
+extern sinsp_evttables g_infotables;
+
+std::string concat_evttypes_names(std::unordered_set<std::string> const names)
 {
-	/// TODO: in the next future we need to change the interface of `enforce_simple_ppm_sc_set`
-	/// and `enforce_sinsp_state_tp` APIs, they shouldn't require an inspector to be called!
-	std::unique_ptr<sinsp> inspector(new sinsp());
+	std::stringstream ss;
+	std::copy(names.begin(), names.end(),
+	std::ostream_iterator<std::string>(ss, ", "));
+	std::string names_str = ss.str();
+	return names_str.substr(0, names_str.size() - 2);
+}
 
-	/* Please note: here we fill these 2 sets because we are interested in only some features, if we leave
-	 * them empty `libsinsp` will fill them with all the available syscalls and all the available tracepoints!
-	 */
+void application::extract_rules_event_names(std::unique_ptr<sinsp>& inspector, std::unordered_set<std::string>& rules_evttypes_names)
+{
+	/* Get all PPME events from all rules as idx codes. */
+	std::set<uint16_t> rule_events;
+	std::string source = falco_common::syscall_source;
+	m_state->engine->evttypes_for_ruleset(source, rule_events);
+	std::unordered_set<uint32_t> ppme_events_codes(rule_events.begin(), rule_events.end());
+
+	/* Translate PPME event codes to event names. */
+	rules_evttypes_names = inspector->get_events_names(ppme_events_codes);
+}
 
-	/* Here the `libsinsp` state set is not enough, we need other syscalls used in the rules,
-	 * so we use the `simple_set`, this `simple_set` contains all the syscalls of the `libsinsp` state
-	 * plus syscalls for Falco default rules.
+void application::activate_interesting_syscalls_events(std::unique_ptr<sinsp>& inspector, const std::unordered_set<std::string>& rules_evttypes_names)
+{
+	/* PART 1: syscalls and events activation
+	 *
+	 * Current enforce_simple_ppm_sc_set approach includes 2 steps:
+	 * (1) Enforce all syscalls from all Falco rules
+	 * (2) Enforce a static set of syscalls that includes all syscalls for the Falco default rules
+	 * (3) Enforce `libsinsp` state set (non-adaptive, not conditioned by rules, but based on PPME event table flags indicating generic sinsp state modifications)
+	 * -> Final set is union of (1), (2) and (3)
+	 *
+	 * New user input ONLY approach via m_state->config->m_additional_syscalls option:
+	 * Additional syscalls of interest providen by the end user in addition to the syscalls/events from all rules. Defaults to EMPTY and
+	 * sinsp state enforcement is applied instead when this option is not set
+	 *
+	 * TODO:
+	 * - Support multiple new more adaptive syscalls activation options as alternatives via new Falco flags to provide maximum control and configurability to end users
```

Comment:
I don't get the TODO :)

Reply:
No worries, I think comments in general need to be improved a few times. Was referring to the options we discussed in  https://github.com/falcosecurity/libs/pull/826/files#r1071797337 (your `sinsp` enforcement method based on EF MODIFIES STATE and the one I proposed which aims to be even more lean). Since we wanted to clean this up in follow up PRs we will get rid of it soon again ;)

Reply:
Aka support both approaches, yours as default and the other one with a new flag.